test/apps/iodemo/run_io_demo.sh

```diff
@@ -560,6 +562,10 @@ make_scripts()
 			            tag="\$2"
 			            shift
 			            ;;
+			        -num_path)
+			            add_cmd_vars+=" UCX_IB_NUM_PATHS=\$2"
```

Comment:
why needed? we could set UCX_IB_NUM_PATHS env var before launching run_io_demo.sh

Reply:
@yosefe 
There're two methods to implement PATH compatible test for io_demo:
- Method 1
transfer ```UCX_IB_NUM_PATHS value``` from [iodemo_run.sh](https://github.com/Mellanox/hpc-mtt-conf/blob/master/scripts/iodemo_run.sh#L387-L389) to ```iodemo_commands``` script created by [run_io_demo.sh](https://github.com/openucx/ucx/blob/master/test/apps/iodemo/run_io_demo.sh) and add ```UCX_IB_NUM_PATHS``` in command line when running io_demo.
This method is implemented in https://github.com/openucx/ucx/pull/7694 & https://github.com/Mellanox/hpc-mtt-conf/pull/732

- Method 2
Make below similar change to set UCX_IB_NUM_PATHS when executing below command in [iodemo_run.sh](https://github.com/Mellanox/hpc-mtt-conf/blob/master/scripts/iodemo_run.sh#L387-L389)
```pdsh -S -b -w ${host_list}  "UCX_IB_NUM_PATHS=${server_num_paths} $LAUNCH_DIR/iodemo_commands_\$(hostname -s).sh -start all_servers" &> /dev/null ```

Which one do you prefer?

Reply:
we should change only in MTT.
@avildema what is the best way to pass different UCX_IB_NUM_PATHS values to io_demo client and server?
can you pls review the MTT PR and suggest?

Reply:
I prefer Method2